### PR TITLE
Add bytes::Bytes implementation

### LINKED ISSUE #259: Include support for bytes::Bytes
Standard crate of the Tokio stack, allows for potentially removing allocations when bytestring is cloned and used verbatim.

Already used in rlp, may as well include support for encoding `Bytes` instances.

### PR DESCRIPTION
**Pull request by vorot93**
Fixes #259

### CODE REVIEW

File: src/bytes.rs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+use crate::{Decode, Encode, EncodeLike, Error, Input};
```
Comment (bkchr) [inline src/bytes.rs:1]: Missing license header. (just copy from one of the other files).

And could you please also add at least some basic test?

**Comment by vorot93** [comment]
@bkchr I reworked the PR to make use of WrapperTypeEncode and EncodeLike, just the way SCALE is implemented for String and memory containers. Also added tests.

**Comment by vorot93** [comment]
Any chance this could be merged soon? @bkchr

File: Cargo.toml [hunk 2]
```diff
@@ -14,6 +14,7 @@ arrayvec = { version = "0.7", default-features = false }
 serde = { version = "1.0.102", optional = true }
 parity-scale-codec-derive = { path = "derive", version = "2.3.1", default-features = false, optional = true }
 bitvec = { version = "0.20.1", default-features = false, features = ["alloc"], optional = true }
+bytes-crate = { package = "bytes", version = "1", default-features = false, optional = true }
```
Comment (bkchr) [inline Cargo.toml:17]:
```suggestion
bytes = { version = "1", default-features = false, optional = true }
```

File: Cargo.toml [hunk 3]
```diff
@@ -39,6 +40,7 @@ default = ["std"]
 derive = ["parity-scale-codec-derive"]
 std = ["serde", "bitvec/std", "byte-slice-cast/std", "chain-error"]
 bit-vec = ["bitvec"]
+bytes = ["bytes-crate"]
```
Comment (bkchr) [inline Cargo.toml:43]:
```suggestion
```

File: src/codec.rs [hunk 4]
```diff
@@ -1429,6 +1446,19 @@ mod tests {
 		assert_eq!(<Vec<OptionBool>>::decode(&mut &encoded[..]).unwrap(), value);
 	}
 
+	#[cfg(feature = "bytes")]
+	#[test]
+	fn bytes_works_as_expected() {
+		let input = bytes_crate::Bytes::from_static(b"hello");
```
Comment (bkchr) [inline src/codec.rs:1452]:
```suggestion
		let input = bytes::Bytes::from_static(b"hello");
```

File: src/codec.rs [hunk 5]
```diff
@@ -1429,6 +1446,19 @@ mod tests {
...
+		let encoded = Encode::encode(&input);
+		let encoded_vec = input.to_vec().encode();
+		assert_eq!(encoded, encoded_vec);
+
+		assert_eq!(
+			bytes_crate::Bytes::decode(&mut (&encoded as &[u8])).unwrap(),
+			Vec::<u8>::decode(&mut (&encoded as &[u8])).unwrap()
```
Comment (bkchr) [inline src/codec.rs:1459]:
```suggestion
			&*b"hello"[..],
			bytes::Bytes::decode(&mut &encoded[..]).unwrap(),
```

File: src/codec.rs [hunk 6]
```diff
@@ -371,6 +371,23 @@ mod feature_full_wrapper_type_encode {
 	impl EncodeLike<String> for &str {}
 }
 
+#[cfg(feature = "bytes")]
+mod feature_wrapper_bytes {
+	use super::*;
+	use bytes_crate::Bytes;
```
Comment (bkchr) [inline src/codec.rs:377]:
```suggestion
	use bytes::Bytes;
```

**Review by bkchr — APPROVED** [review]
Sorry for the delay

File: src/codec.rs [hunk 7]
```diff
@@ -1429,6 +1446,19 @@ mod tests {
...
+	#[cfg(feature = "bytes")]
+	#[test]
+	fn bytes_works_as_expected() {
+		let input = bytes::Bytes::from_static(b"hello");
+		let encoded = Encode::encode(&input);
+		let encoded_vec = input.to_vec().encode();
+		assert_eq!(encoded, encoded_vec);
+
+		assert_eq!(
+			&*b"hello"[..],
```
Comment (bkchr) [inline src/codec.rs:1458]:
```suggestion
			&b"hello"[..],
```

**Comment by bkchr** [comment]
Ty for fixing the ui test!

**Comment by shawntabrizi** [comment]
@vorot93 if you post your polkadot or kusama address, I will open a treasury tip for you

**Comment by vorot93** [comment]
@shawntabrizi Sure

`15rc4P2NQ3zuzk2r3RYyYtYmCV7aeMGbRdEHxrT2UNdJQpfr` on Polkadot

Thanks!

**Comment by shawntabrizi** [comment]
Done: https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Frpc.polkadot.io#/treasury/tips

**Pull request merged**